x/evm/keeper/grpc_query.go

```diff
@@ -426,22 +426,24 @@ func (k Keeper) TraceTx(c context.Context, req *types.QueryTraceTxRequest) (*typ
 
 	txConfig := statedb.NewEmptyTxConfig(common.BytesToHash(ctx.HeaderHash().Bytes()))
 	for i, tx := range req.Predecessors {
-		ethTx := tx.AsTransaction()
-		msg, err := ethTx.AsMessage(signer, cfg.BaseFee)
+		coreMsg, err := tx.AsMessage(signer, cfg.BaseFee)
```

Comment:
we need to use `MsgEthereumTx.AsMessage` instead of `ethtypes.Transaction.AsMessage` so we can replace the sender